Lib/test/test_capi/test_unicode.py

```diff
@@ -1297,6 +1297,54 @@ def test_comparewithasciistring(self):
         # CRASHES comparewithasciistring([], b'abc')
         # CRASHES comparewithasciistring(NULL, b'abc')
 
+    @support.cpython_only
+    @unittest.skipIf(_testcapi is None, 'need _testcapi module')
+    def test_equaltoutf8(self):
+        """Test PyUnicode_EqualToUTF8()"""
+        from _testcapi import unicode_equaltoutf8 as equaltoutf8
+        from _testcapi import unicode_asutf8andsize as asutf8andsize
+
+        strings = [
+            'abc', '\xa1\xa2\xa3', '\u4f60\u597d\u4e16',
+            '\U0001f600\U0001f601\U0001f602'
```

Comment:
Maybe test also MAX_UNICODE explicitly:

```suggestion
            '\U0001f600\U0001f601\U0001f602',
            '\U0010ffff',
```